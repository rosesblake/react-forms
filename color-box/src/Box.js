import React, { useState } from "react";

const Box = ({ color, width, height, deleteBox, box }) => {
  return (
    <div
      data-testid="box"
      className="Box"
      style={{
        backgroundColor: color,
        width: `${width}px`,
        height: `${height}px`,
      }}
    >
      <button onClick={() => deleteBox(box)}>X</button>
    </div>
  );
};

export default Box;
