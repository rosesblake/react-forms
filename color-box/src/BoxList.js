import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const renderBoxes = () => {
    return (
      <>
        {boxes.map((box) => (
          <Box
            key={box.id}
            color={box.color}
            width={box.width}
            height={box.height}
            deleteBox={deleteBox}
            box={box}
          />
        ))}
      </>
    );
  };

  const addBox = (box) => {
    let newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };

  const deleteBox = (box) => {
    setBoxes((boxes) => boxes.filter((b) => b.id !== box.id));
  };

  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      {renderBoxes()}
    </div>
  );
};

export default BoxList;
