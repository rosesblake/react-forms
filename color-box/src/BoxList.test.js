import React from "react";
import { render, fireEvent, screen, within } from "@testing-library/react";
import BoxList from "./BoxList";
import Box from "./Box";

it("renders", () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <BoxList>
      <Box color="blue" width={10} height={10} />
    </BoxList>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("updates based off of form data", () => {
  render(<BoxList />);

  const colorInput = screen.getByLabelText("Color:");
  const widthInput = screen.getByLabelText("Width:");
  const heightInput = screen.getByLabelText("Height:");
  const button = screen.getByRole("button");

  fireEvent.change(colorInput, { target: { value: "black" } });
  fireEvent.change(widthInput, { target: { value: 10 } });
  fireEvent.change(heightInput, { target: { value: 10 } });

  fireEvent.click(button);

  const box = screen.getByTestId("box");
  expect(box).toBeInTheDocument();

  expect(box.style._values).toEqual({
    "background-color": "black",
    width: "10px",
    height: "10px",
  });
});
