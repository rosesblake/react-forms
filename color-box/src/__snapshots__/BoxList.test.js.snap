// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <h1>
        Create New Box
      </h1>
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        value=""
      />
      <br />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="pixels"
        value=""
      />
      <br />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="pixels"
        value=""
      />
      <br />
      <button>
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
